pkg/api/types.go

```diff
@@ -1801,6 +1801,9 @@ type PodSpec struct {
 	// ServiceAccountName is the name of the ServiceAccount to use to run this pod
 	// The pod will be allowed to use secrets referenced by the ServiceAccount
 	ServiceAccountName string
+	// AutomountServiceAccountToken indicates whether a service account token should be automatically mounted.
+	// +optional
+	AutomountServiceAccountToken *bool
```

Comment:
All I can think of is setting a volume with a special name but not mounting it.  But from a security perspective we want to ACL secrets according to references, not use, so that would potentially expose the token still.